concordium-contracts-common/src/types.rs

```diff
@@ -1406,6 +1536,67 @@ pub struct Policy<Attributes> {
     pub items:             Attributes,
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+fn gen_no_dup_kv_vec<A: quickcheck::Arbitrary + Ord, B: quickcheck::Arbitrary>(
+    g: &mut Gen,
+    size: usize,
+) -> Vec<(A, B)> {
+    use std::collections::BTreeMap;
+    let mut m: BTreeMap<A, B> = BTreeMap::new();
+    while m.len() < size {
+        let k = A::arbitrary(g);
+        let v = B::arbitrary(g);
+        m.insert(k, v);
+    }
+    m.into_iter().collect()
```

Comment:
Perhaps it would make sense to relax this to avoid infinite loops. And just generate vectors of size <= m.

```
for _ in 0..m {

        let k = A::arbitrary(g);
        let v = B::arbitrary(g);
        m.insert(k, v);
}
```

Reply:
Done.